Nix-CAS rfc

thufschmitt opened the pull request:
A draft PR to gather some preliminary reviews on this RFC before (hopefully) submitting it upstream.

This ist still quite WIP and probably needs a lot of rewriting, but the bulk of the content should be there

thufschmitt committed:
CAP RFC: First draft

infinisil committed:
[RFC 0052] Away from static IDs (#52)

aspiwack commented:
I had a few questions for @regnat which he asked me to reproduce here for safekeeping:

- “rewrite it” or similar phrases is used throughout the document, but really, if I understand correctly, nothing is being rewriten: we compute `dynamic(derivation)`, then realise it. Nothing is replaced by anything.
- What's in the store at the static name of a derivation?
- Why do you choose to explicitly mark something as content-addressed?
   - Turns out it's a purely engineering consideration, a means to an end, not a staple of the design, in fact, it's only transitory. This should be made clear.
- How does the design compare with other RFCs? Is the goal to eventually converge to the same design, or is it a new design
  - It's the essentially same as https://github.com/NixOS/rfcs/pull/17 . Only the path is different. If I understood correctly.

The answer to all of these needs to make it in the RFC document before it is proposed to the community.

picnoir commented on rfcs/0050-content-addressed-paths.md:
Is `if its build is not deterministic` == FOD here?

If not, how do you assume a build is not deterministic?

thufschmitt replied:
That depends on how we end-up implementing it, but it's reasonably simple to implement something like "your local build of `foo` differs from the one on the remote cache => You get cache misses for everything that depends on `foo`" (so some form of local-only FOD)

ryantm committed:
[RFC 0051] Mark stale nixpkgs issues (#51)

* rfc51: close stale issues initial draft

* rfc0051: use staleLabel consistent with nixpkgs labeling conventions

* rfc0051: do not close issues

* rfc0051: reword motivation

* record the members of the shepherd team

* Update rfcs/0051-mark-stale-issues.md

Co-Authored-By: asymmetric <lorenzo@mailbox.org>

* [RFC 0051] remove references to "issue" in stale message

curiousleo committed:
README.md: remove template paragraph (#61)

This paragraph seems to be a leftover or copy-pasta from the RFC template. It doesn't really fit into the flow of the text, so I think it was included or left here by mistake.

Ericson2314 committed:
Make the detailed design terse by separating examples (#60)

7c6f434c committed:
[RFC 0045] Deprecating unquoted URL syntax (#45)

* A small RFC on deprecating URL syntax

* Convert alternatives to a list

* Add a mention of tooling in future work

* A remark from @globin about one more problem with unquoted URLs

* Grammar edit: comparison of URLs, paths and strings.

Co-Authored-By: 7c6f434c <7c6f434c@mail.ru>

* Style edit: possibility of future removal

Co-Authored-By: 7c6f434c <7c6f434c@mail.ru>

* 0045: commit to using the tooling that now exists

* Update rfcs/0045-deprecate-url-syntax.md

Shepherd team

Co-Authored-By: Domen Kožar <domen@enlambda.com>

* Fix shepherd list

* Update based on a discussion

* Explicitly make future removal conditional on editions; restrict the claim about no special support to the Nix language itself

thufschmitt committed:
typo

Commit:
Apply @grahamc's suggestions

Co-Authored-By: Graham Christensen <graham@grahamc.com>

thufschmitt committed:
nix code -> Nix expression

thufschmitt committed:
Break-up the big introduction paragraph

As suggested in https://github.com/NixOS/rfcs/pull/62/files#r356694585

thufschmitt committed:
Rename to match the PR number

thufschmitt committed:
Rename the drv attribute to __contentAddressed

Makes it more in line with other "magic" attributes like
`__structuredAttributes`

Also fix the orthograph

aspiwack commented on rfcs/0062-content-addressed-paths.md:
```suggestion
critical derivations as content-adressed can lead to some substantial build
```

aspiwack commented on rfcs/0062-content-addressed-paths.md:
Are you able to document (with links, presumably) some of the unsolved implications?

aspiwack commented on rfcs/0062-content-addressed-paths.md:
And storage reduction on hydra, and less download on updates.

aspiwack commented on rfcs/0062-content-addressed-paths.md:
Still “rewritten”? I'm still not convinced.

aspiwack commented on rfcs/0062-content-addressed-paths.md:
This paragraph is information free (like the broccoli example) I suggest complete removal.

```suggestion
```

aspiwack commented on rfcs/0062-content-addressed-paths.md:
```suggestion
```

aspiwack commented on rfcs/0062-content-addressed-paths.md:
```suggestion
Consider the following Nix expression:
```

aspiwack commented on rfcs/0062-content-addressed-paths.md:
Footnotes don't render in Github :slightly_frowning_face:

aspiwack reviewed (commented):
A handful of comments

jonas-schievink committed:
template: fix typo (#63)

thufschmitt committed:
Mention the GC issue

thufschmitt committed:
Remove the ambiguity on what an `output` is

7c6f434c committed:
[RFC 0046] Platform Support Tiers (#46)

[RFC 0046] Platform Support Tiers

thufschmitt committed:
Replace aliases paths by a pathOf mapping

thufschmitt committed:
Move the example after the design description

thufschmitt committed:
Rephrase the design

In particular, replace `static` and `dynamic` by `symbolic` and
`resolved`

thufschmitt committed:
Add shepherd team

worldofpeace committed:
Update members of RFC Steering Comittee (#65)

thufschmitt committed:
Rewrite the RFC to account for the RFC meeting comments

- Add the notion of `drvOutputId`
- Replace the alias paths by a `PathOf(drvOutputId)` function
- Mention the notion of "truster" in the `PathOf` function

thufschmitt committed:
Add a section about leaking output paths

Ericson2314 committed:
Merge remote-tracking branch 'upstream/master' into cas-rfc

Commit:
Merge pull request #5 from Ericson2314/cas-rfc-new-template

content-addressed derivations RFC: merge in new master

thufschmitt committed:
Refine the design summary

thufschmitt committed:
Rename dependency-addressed into input-addressed

And define the term at the begining of the RFC

thufschmitt committed:
minor fixup after comments

Commit:
Apply suggestions from code review

Co-authored-by: asymmetric <lorenzo@mailbox.org>
Co-authored-by: Profpatsch <philip.patsch@tweag.io>

Mic92 committed:
Update rfcs/0062-content-addressed-paths.md

thufschmitt committed:
Update the terminology to match the in the implementation

thufschmitt committed:
Reword the detailed design presentation

thufschmitt committed:
Quote some strings in the yaml frontmatter

thufschmitt committed:
Add a design paragraph about the remote caching

And update the “unresolved questions” to take it into account

thufschmitt committed:
Lift the determinism requirement

It's not required anymore by the current remote caching semantics

edolstra committed:
Typo

Co-authored-by: davidak <davidak@users.noreply.github.com>

edolstra committed:
Apply suggestions from code review

Co-authored-by: davidak <davidak@users.noreply.github.com>

thufschmitt committed:
Rewrite the RFC

This is mostly a full-rewrite of the RFC to

1. Make it more “incremental”: A first part just describes the minimal
   model upon which everything is based, and a second one shows
   different extensions of this model to add more features
2. Remove the big ugly examples that don’t add much value because they
   aren’t really readable
3. Add a python pseudo-code pseudo-formalisation of the RFC.
  This is imho both more readable and precise than nested bullet-points
  of handwaved language

Commit:
Make the python samples a bit more pythonic

Co-authored-by: zseri <zseri.devel@ytrizja.de>

thufschmitt committed:
Explicit that unresolved dependencies are eval-time

thufschmitt committed:
Prettify

Commit:
Make the end-goal an experiment

Simplify the paperwork and just get this to FCP because right now it’s stuck in a hole
